sbncode/FlashMatch/FlashPredict_module.cc

```diff
@@ -247,78 +252,97 @@ void FlashPredict::produce(art::Event& evt)
+        sFM_v->emplace_back(sFM(kNoScr, kNoScrTime, Charge(kNoScrQ),
+                                Flash(kNoScrPE), Score(kNoOpHInEvt)));
+        util::CreateAssn(*this, evt, *sFM_v, pfp_ptr, *pfp_sFM_assn_v);
+      }
+      evt.put(std::move(sFM_v));
+      evt.put(std::move(pfp_sFM_assn_v));
+      return;
     }
-    evt.put(std::move(sFM_v));
-    evt.put(std::move(pfp_sFM_assn_v));
-    return;
-  }
-
-  std::vector<recob::OpHit> opHits(ophits_h->size());
-  copyOpHitsInFlashFindingWindow(opHits, ophits_h);
 
-  if(fUseARAPUCAS && !fOpHitARAProducer.empty()){
-    art::Handle<std::vector<recob::OpHit>> ophits_ara_h;
-    evt.getByLabel(fOpHitARAProducer, ophits_ara_h);
-    if(!ophits_ara_h.isValid()) {
-      mf::LogError("FlashPredict")
-        << "Non valid ophits from ARAPUCAS"
-        << "\nfUseARAPUCAS: " << std::boolalpha << fUseARAPUCAS
-        << "\nfOpHitARAProducer: " << fOpHitARAProducer;
+    opHits.resize(ophits_h->size());
+    copyOpHitsInFlashFindingWindow(opHits, ophits_h);
+  
+    mf::LogInfo("FlashPredict")
+      << "OpHits found: " << opHits.size() << std::endl;
+
+    const std::vector<SimpleFlash> simpleFlashes_tmp = (fSBND) ?
+      makeSimpleFlashes(opHits, opHitsRght, opHitsLeft) : makeSimpleFlashes(opHits);
+    auto is_flash_in_time = [this](const SimpleFlash& f) -> bool
+    { return (fBeamSpillTimeStart<=f.maxpeak_time &&
+              f.maxpeak_time<=fBeamSpillTimeEnd); };
+    auto flash_in_time = std::find_if(simpleFlashes_tmp.begin(), simpleFlashes_tmp.end(),
+                                      is_flash_in_time);
+
+    if(simpleFlashes_tmp.empty() ||
+       flash_in_time == simpleFlashes_tmp.end()){
+      mf::LogWarning("FlashPredict")
+        << "No SimpleFlashes in beam window [" << fBeamSpillTimeStart << ", " << fBeamSpillTimeEnd << "], "
+        << "\nor the sum of PE is less or equal to " << fMinFlashPE << " or 0."
+        << "\nSkipping...";
+      bk.nullophittime++;
+      updateBookKeeping();
+      for(size_t pId=0; pId<pfps_h->size(); pId++) {
+        if(!pfps_h->at(pId).IsPrimary()) continue;
+        const art::Ptr<recob::PFParticle> pfp_ptr(pfps_h, pId);
+        sFM_v->emplace_back(sFM(kNoScr, kNoScrTime, Charge(kNoScrQ),
+                                Flash(kNoScrPE), Score(kNoOpHInEvt)));
+        util::CreateAssn(*this, evt, *sFM_v, pfp_ptr, *pfp_sFM_assn_v);
+      }
+      evt.put(std::move(sFM_v));
+      evt.put(std::move(pfp_sFM_assn_v));
+      return;
+    } else {
+    std::copy(simpleFlashes_tmp.begin(), simpleFlashes_tmp.end(), std::back_inserter(simpleFlashes));
+    for(auto& sf : simpleFlashes) flashMetrics.push_back(getFlashMetrics(sf));
     }
-    else{
-      std::vector<recob::OpHit> opHitsARA(ophits_ara_h->size());
-      copyOpHitsInFlashFindingWindow(opHitsARA, ophits_ara_h);
-      opHits.insert(opHits.end(),
-                    opHitsARA.begin(), opHitsARA.end());
+  } else {
+    std::vector<FlashMetrics> flashMetrics_tmp;
```

Comment:
It doesn't seem to me like you need this extra variable `flashMetrics_tmp`, you can simply push back on the `flashMetrics` object if the checks pass. This would save you from needing to copy.